### PR TITLE
Add Apache2 License

### PR DESCRIPTION
**Pull request by pmclanahan**
I decided on Apache2 over MPL2 since it's less restrictive and non-copyleft. Since this seems more like a tool than a site this seems better to me.

@Osmose @glogiotatidis r?

### CODE REVIEW

**Comment by Osmose** [comment]
A line in the README mentioning the license would be useful. Otherwise r+ from me (but @glogiotatidis 's feedback should be required before merging)

**Comment by pmclanahan** [comment]
Added README license mention.

**Comment by peterbe** [comment]
Perhaps a line in [setup.py](https://github.com/mozilla/sugardough/blob/master/%7B%7B%20cookiecutter.project_name%20%7D%7D/setup.py) so it shows up the LICENSE on PyPI would be nice too.

**Comment by pmclanahan** [comment]
@peterbe I don't see a `setup.py` in this repo.

**Comment by peterbe** [comment]
https://github.com/mozilla/sugardough/blob/master/%7B%7B%20cookiecutter.project_name%20%7D%7D/setup.py

**Comment by glogiotatidis** [comment]
OK that's an interesting one:

We're having a tool `sugardough` that generates websites. In this pull request it's suggested that the tool gets Apache2 licensed. The tool generates [MPLv2](https://github.com/mozilla/sugardough/blob/master/%7B%7B%20cookiecutter.project_name%20%7D%7D/LICENSE) licensed websites (btw @peterbe the setup.py is part of the generated website)

Practically since sugardough is a template tool targeting mozilla websites it makes sense. Mozilla websites should be MPLv2 licensed. 

Does having a tool licensed X to generate code licensed Y make sense?

**Comment by peterbe** [comment]
Crap! I've told myself to almost never get involved in licensing because I'm a licensing-idiot. 

I see what you mean. There's two different licenses here. One for sugardough (the generator) and one for the template (the generated). 

@pmclanahan You seem to know a lot about licensing. Should we start with just apache for sugardough and keep MPL2 for the template? 

Either way, I do know that the setup.py should include **a** license but perhaps that can wait.

**Comment by pmclanahan** [comment]
This is more interesting than I thought :) I think it'd be fine to have the tool use one license and suggest another for the generated site. I agree with @peterbe that the generated setup.py should include licensing information, but that's a separate issue and I think should be done outside of this PR to avoid confusion.

**Comment by pmclanahan** [comment]
Also, to be clear, I do like by default using MPL2 for the generated sites.

**Comment by pmclanahan** [comment]
Also also, after I found that there was no setup.py for sugardough I started thinking about how we might use one and whether having this on PyPI would be good. I know it requires cookiecutter, but that dependency could be handled in the setup.py and potentially run on the template as part of install. My thought is something like:

1. `pip install sugardough`
2. questions are asked and the site is generated 
3. build site
4. profit!

Thoughts? If this seems at all useful and/or feasible I'll file a new issue about this.

**Comment by glogiotatidis** [comment]
So sugardough gets Apache2 licensed and sugardough based projects are MPL. that lgtm. Unless there is an objection i'll merge this tomorrow.

> Thoughts? If this seems at all useful and/or feasible I'll file a new issue about this.
+1 setup.py for sugardough sounds neat.

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {yyyy} {name of copyright owner}
```
Comment (peterbe) [inline LICENSE]: Don't you need to fill this in?

File: README.md [hunk 2]
```diff
@@ -107,3 +107,8 @@ For test coverage you'll also have to add `pytest-cov`.
 Next you'll need to edit the `.travis.yml` file and edit the script part.
 Instead of `coverage run manage.py test` it
 `py.test --cov=sugardough`.
+
+License
+-------
+
+Sugardough is licensed under the Apache 2 license. See the `LICENSE` file in this repository for the full text of the license.
```
Comment (peterbe) [inline README.md]: Can you change that to `[LICENSE](LICENSE)` to make it a link?

File: LICENSE (hunk 1 above)
Comment (pmclanahan) [inline LICENSE]: HA! Good catch.

Comment (pmclanahan) [inline LICENSE]: Hmm... actually this is a section just talking about how to apply the license to your files and specifically references the brackets. A copyright notice is likely a good idea somewhere though.

**Comment by glogiotatidis** [comment]
r+ thanks! :dancers:

**Pull request merged**